frontend/manage_user.tsx

```diff
@@ -0,0 +1,220 @@
+/**
+ * @file This file contains the ManageUser function. This function is used for
+ * add users and set user permissions.
+ */
+import React from 'react'
+import { MenuItem, InputLabel, Select, TextField, Checkbox, FormControlLabel, Button } from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+import { get, post, update } from './communication';
+import { useDataContext } from './data_context'
+import { Genebank, Herd } from '~data_context_global';
+
+// Define styles for tab menu
+const useStyles = makeStyles({
+  form: {
+    borderLeft: "1px solid rgba(0,0,0,0.1)",
+    paddingLeft: "0.5cm",
+    display: "flex",
+    flexDirection: "column",
+  },
+  simpleField: {
+    width: "400px",
+  },
+});
+
+/** description of a user to manage */
+export interface ManagedUser {
+  id: number
+  email: string
+  validated: boolean
+  privileges: any
+  password?: string
+}
+
+/**
+ * Provides a form for changing user metadata and user roles.
+ */
+export function ManageUser({id}: {id: number | string | undefined}) {
```

Comment:
Here `id: number | 'new' | undefined` should probably not be called id...

Maybe split this into two components `ManageUser` and `CreateNewUser` and call the correct one from `ManageUsers`?